### PR TITLE
Release: v1.0.1

### PR DESCRIPTION
**Pull request by sgammon**
This changeset adds support for numerous schema changes, and makes improvements with the goal of thinning down the library to a smaller runtime size.

- [x] Version bump -> ~~`v1.0.1-rc2`~~ `v1.0.1-rc4`
- [x] Remove dependency on Closure UI
- [x] Tune up closure flags and clean up various things
- [x] Adapt to schema v1.5.0
  - [x] Use new product key structures
  - [x] Use new weight structures
- [x] Publish beta version of lib for testing

Blocking downstream:
- bloombox/shop-beta#12: Shop Data v2

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -20,6 +20,14 @@
       "source" : "/latest-debug.min.js",
       "destination" : "/client/v1.0.1-rc1-debug.min.js",
       "type" : 302
+    },{
+      "source" : "/beta.min.js",
+      "destination" : "/client/v1.0.1-rc2.min.js",
+      "type" : 302
+    },{
+      "source" : "/beta-debug.min.js",
```
Comment (sgammon) [inline firebase.json]: new beta libs means you can test against `https://js.bloombox.cloud/beta.min.js` and `../beta-debug.min.js`

File: gulpfile.js [hunk 2]
```diff
@@ -86,6 +91,9 @@ function closureBuilder(entrypoint) {
       "rewrite_polyfills": true,
       "D": "bloombox.DEBUG",
       "define": "bloombox.VERSION='" + version + "'",
+      "create_source_map": "target/js-" + entrypoint + ".map",
+      "output_module_dependencies": "target/deps-" + entrypoint + ".json",
+      "source_map_include_content": true,
```
Comment (sgammon) [inline gulpfile.js]: we're gonna have source maps support soon

File: src/base/init.js [hunk 3]
```diff
@@ -29,9 +29,6 @@ goog.provide('bloombox.DEBUG_PROPERTY');
 goog.provide('bloombox.INTERNAL');
 goog.provide('bloombox.VERSION');
 
-// force-load closure and closure UI libraries
-goog.require('goog.ui.INLINE_BLOCK_CLASSNAME');
```
Comment (sgammon) [inline src/base/init.js]: no need to depend on closure UI libs anymore, yay

File: src/product/product.js [hunk 4]
```diff
@@ -31,8 +31,7 @@ goog.require('bloombox.util.Exportable');
 
 goog.require('proto.opencannabis.base.ProductKey');
 goog.require('proto.opencannabis.base.ProductKind');
-goog.require('proto.opencannabis.base.ProductType');
-goog.require('proto.opencannabis.commerce.ProductWeight');
+goog.require('proto.opencannabis.structs.pricing.PricingWeightTier');
```
Comment (sgammon) [inline src/product/product.js]: adapt to new key structures and weights

File: firebase.json [hunk 5]
```diff
@@ -20,6 +20,14 @@
       "source" : "/latest-debug.min.js",
       "destination" : "/client/v1.0.1-rc1-debug.min.js",
       "type" : 302
```
Comment (RandyStevens) [inline firebase.json]: Should we remove this rc1 code here?

Comment (sgammon) [reply firebase.json]: Nah, we remove that when it goes live. it's in "beta" right now, below.

Comment (sgammon) [reply firebase.json]: Good thought though

Comment (RandyStevens) [reply firebase.json]: 👍

**Comment by sgammon** [comment]
after testing in prod and review by @noahdamiani we can merge

**Comment by noahdamiani** [comment]
It is midnight but I can confirm that RC4 has everything needed for the current state of shop. Everything is switched over on my end as of my recent commit (shop-beta/pull/12), so we should be good to merge!

**Pull request merged**